spec/integrations/buffering_spec.rb

```diff
@@ -28,6 +28,20 @@
     expect(socket.recv).to be_nil
   end
 
+  it 'sends multiple samples in one packet with the batch compatility method' do
+    subject.batch do |s|
+      s.increment('mycounter')
+      s.increment('myothercounter')
+      s.decrement('anothercounter')
+      s.increment('myothercounter')
+      s.decrement('yetanothercounter')
+    end
+    # multiple reads since buffer_max_pool_size == 2
```

Comment:
Maybe for this test you could put a way bigger `max_pool_size` and a bigger `max_buffer_size` (so they do not trigger flushes) and then expect all the batch at once. WDYT?

Reply:
Adding a test to test this as well 👍